### PR TITLE
Added more serializer/deserializer tests.

### PR DESCRIPTION
**Pull request by dblock**
### Description

Coming from https://github.com/opensearch-project/opensearch-js/pull/558.  Are there cases that would still fail?

### Check List

- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] Linter check was successfull - `yarn run lint` doesn't show any errors
- [ ] Commits are signed per the DCO using --signoff
- [ ] Changelog was updated.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: test/unit/serializer.test.js [hunk 1]
```diff
@@ -55,7 +55,11 @@ test('Long numerals', (t) => {
     `"positive": ${longPositive.toString()}, ` +
     `"array": [ ${longNegative.toString()}, ${longPositive.toString()} ], ` +
     `"negative": ${longNegative.toString()},` +
-    `"hardcoded": 102931203123987` +
+    `"hardcoded": 102931203123987,` +
+    `"a": "෴1111111111111111",` +
+    `"b": "߷1111111111111111",` +
+    `"c": "֍1111111111111111",` +
+    `"d": "෴֍߷1111111111111111"` +
     `}`;
```
Comment (nhtruong) [inline test/unit/serializer.test.js:63]: Don't you want to add the expected outputs like `t.equal(obj.hardcoded, 102931203123987);` on line 71?

Comment (AMoo-Miki) [reply test/unit/serializer.test.js:63]: I will be adding similar ones in my other PR. I would recommend we hold off on these.

Comment (dblock) [reply test/unit/serializer.test.js:65]: Updated.

**Comment by dblock** [comment]
@AMoo-Miki feel free to close this one if you have better

**Comment by dblock** [comment]
https://github.com/opensearch-project/opensearch-js/pull/557 is better.

**Pull request closed without merging**